File: ios/RensNativeViewManager.m
//
//  RensNativeViewManager.m
//  Rens
//
//  Created by Ivan Driuk on 1/15/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RensNativeViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(count, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(onUpdate, RCTDirectEventBlock)
RCT_EXTERN_METHOD(
                  updateFromManager:(nonnull NSNumber *)node
                  count:(nonnull NSNumber *)count
                  )
@end

File: ios/RensNativeViewManager.swift
//
//  RensNativeViewManager.swift
//  Rens
//
//  Created by Ivan Driuk on 1/15/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

@objc(RensNativeViewManager)
class RensNativeViewManager: RCTViewManager {
  override func view() -> UIView! {
    return RensNativeView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  func updateFromManager(_ node: NSNumber, count: NSNumber) {
    
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
        ) as! RensNativeView
      component.update(value: count)
    }
  }
  
}
